lib/semian/simple_state.rb

```diff
@@ -0,0 +1,49 @@
+module Semian
+  module Simple
+    class State #:nodoc:
+      def initialize
+        @enums = [:closed, :open, :half_open]
+        reset
+      end
+
+      attr_reader :value
+
+      def value=(val)
+        raise ArgumentError unless @enums.include?(val)
+        @value = val
+      end
```

Comment:
So then, if I remove this `test_will_throw_error_when_invalid_symbol_given` will be gone as well?